gix-ref/fuzz/fuzz_targets/fuzz_packed_buffer.rs

```diff
@@ -0,0 +1,54 @@
+#![no_main]
+
+use anyhow::Result;
+use arbitrary::Arbitrary;
+use bstr::BStr;
+use gix_ref::{packed::Buffer, FullNameRef, PartialName};
+use libfuzzer_sys::fuzz_target;
+use std::fs;
+use std::hint::black_box;
+use std::path::PathBuf;
+
+struct FuzzedBufferPath(PathBuf);
+
+impl FuzzedBufferPath {
+    fn new() -> Self {
+        Self(PathBuf::from(format!("/tmp/fuzz_packed_buffer_{}", std::process::id())))
```

Comment:
I think this really must be `tempfile::NamedTempFile` or else it will be racy (or might become racy) depending on the way the fuzzer parallelises. Right now it assumes process ids are enough and I don't know if that's always going to be true.

For safety, using clash-free tempfile seems like the better choice.

Reply:
> How do you feel about me refactoring [open](https://github.com/Byron/gitoxide/blob/0fe20e80145419e1662f869657dabf689786395f/gix-ref/src/store/packed/buffer.rs#L33) into two functions one that opens from a `&[u8]` and is memory backed and the other which maintains the same api i.e. dynamically chooses memory Mapped/InMemory backing.

A great idea! This is the way to go :).

Reply:
Done.